save-deploy/README.md

```diff
@@ -78,6 +78,110 @@ Usually not the whole stack is required for development. Application logic is pe
   `docker compose up -d dex` and then start `api-gateway` with `dev` profile enabled. Using [`application-dev.yaml`](../api-gateway/src/main/resources/application-dev.yml)
   one can connect gateway with dev build of frontend running with webpack by changing `gateway.frontend.url`.
 
+### Using OAuth with a local deployment
+
+ * When the default [`dev-server.js`](../save-frontend/webpack.config.d/dev-server.js)
+   is used, the front-end is expected to communicate directly with the back-end, 
+   omitting any gateway. When enabling OAuth, make sure the gateway is contacted
+   instead:
+ 
+   * `context`: add `/sec/**`, `/oauth2/**`, and `/login/oauth2/**` to the list;
+   * `target`: change to [`http://localhost:5300`](http://localhost:5300) (the
+     default gateway URL); 
+   * `onProxyReq`: drop the entire callback, since both headers (`Authorization`
+     and `X-Authorization-Source`) will be set by the gateway now (the gateway
+     acts as a reverse proxy);
+   * `bypass`: drop the entire callback.
+
+   The resulting `dev-server.js` should look like this:
+   ```javascript
+   config.devServer = Object.assign(
+       {},
+       config.devServer || {},
+       {
+         proxy: [
+           {
+             context: ["/api/**", "/sec/**", "/oauth2/**", "/login/oauth2/**", "**.ico", "**.png"],
+             target: 'http://localhost:5300',
+             logLevel: 'debug',
+           }
+         ]
+       }
+   )
+   ```
+ * Avoid potential name conflicts between local users (those authenticated using
+   _HTTP Basic Auth_) and users created via an external _OAuth_ provider. For
+   example, if you have a local user named `torvalds`, don't try to authenticate
+   as a [_GitHub_ user with the same name](https://github.com/torvalds).
+
+#### _Dex_-specific notes
+
+ * In order to run _Dex_, you need a `build/docker-compose.yaml` file generated.
+   This is done by running  
+   ```bash
+   ./gradlew generateComposeFile
+   ```
+ * The YML configuration file, [`dex.dev.yaml`](dex.dev.yaml), has a syntax
+   explained [here](https://github.com/dexidp/dex/blob/master/examples/config-dev.yaml)
+   and [here](https://github.com/wearearima/spring-boot-dex/blob/master/dex/spring-boot-demo.yaml).
+ * More users can be added using a static configuration via `dex.dev.yaml`.
+   Essential fields explained:
+ 
+   * `hash`: the `bcrypt` hash of the password string:
+     ```bash
+     echo 'password' | htpasswd -BinC 16 'user' | cut -d: -f2
+     ```
+     The `htpasswd` utility is a part of `apache2-utils` package. The maximum
+     cost supported by `htpasswd` is **17**. _Dex_, on the other hand, only
+     allows values up to **16**.
+   * `username`: this is the name of the user from _Dex_ perspective only. Since
+     _Dex_ (unlike _GitHub_), provides no means to query user details (i.e.
```

Comment:
According to [this PR id Dex](https://github.com/dexidp/dex/pull/1473/files) it seems that there should be a `/userinfo` endpoint

Reply:
Thanks, will look into it later. 

I'll definitely document it if I make any progress.